lib/pbench/common/selinux.py

```diff
@@ -0,0 +1,45 @@
+"""SELinux wrapper module.
+
+There is no "pip" installable `selinux` python module, just a shim, of sorts [1],
+which provides a way to add the installed system's `selinux` module to a
+virtualenv.  But that does not work when the virtualenv uses a version of
+python different from the python version where the real `selinux` module is
+installed.
+
+The only time we need a fully working `selinux` module is when we are not
+running tests.
+
+This wrapper acts as a pass-through to the real `selinux` module for the methods
+we need, when that module is present.  When that module is not present, we
+create a class to mimic the behaviors we need.  That created class declares that
+SELinux is not installed (since we don't have a real `selinux` module), and
+provides methods to help us detect logic errors.
+
+[1] https://github.com/pycontribs/selinux
+"""
+try:
+    import selinux
+except ImportError:
+
+    class selinux:
+        """Pbench SELinux wrapper."""
+
+        @staticmethod
+        def is_selinux_enabled():
+            """Always returns 0."""
+            return 0
+
+        @staticmethod
+        def restorecon(path):
+            """Always raises an exception to identify logic problems."""
+            raise Exception(
+                "Logic bomb!  selinux.restorecon() called when selinux is not enabled."
+            )
+
+
+def is_selinux_enabled():
+    return selinux.is_selinux_enabled()
```

Comment:
It bothers me slightly that sometimes here `selinux` will be a package and sometimes it will be a class.  It's an elegant hack that Python doesn't care, but I wonder if maybe we shouldn't do better.

For instance, it wouldn't be hard to set up functions/wrappers/pointers in each of the `try` and `except` branches at lines 20-22, where we know what's what.

Reply:
For instance, I think something like this would be better:
```
try:
    import selinux

    # Forward  our functions to the real implementations.                                                                                                                                                                             
    is_selinux_enabled = selinux.is_selinux_enabled
    restorecon = selinux.restorecon
except ImportError:
    # The import failed; provide stub implementations.

    def is_selinux_enabled():
        """Always indicates "disabled"."""
        return 0

    def restorecon(path):
        """Always raises an exception, to identify logic problems."""
        raise NotImplementedError(
            "Logic bomb!  selinux.restorecon() called when selinux is not enabled."
        )
```
or
```
try:
    import selinux
except ImportError:
    # The import failed; provide stub implementations.

    def is_selinux_enabled():
        """Always indicates "disabled"."""
        return 0

    def restorecon(path):
        """Always raises an exception, to identify logic problems."""
        raise NotImplementedError(
            "Logic bomb!  selinux.restorecon() called when selinux is not enabled."
        )

else:
    # The import succeeded; forward our functions to the real implementations.
    is_selinux_enabled = selinux.is_selinux_enabled
    restorecon = selinux.restorecon
```

Reply:
I took the second suggestion.

Reply:
Ha!  You and your affinity for `else` clauses on `try` blocks.  😀